gammapy/maps/wcs.py

```diff
@@ -741,31 +741,63 @@ def separation(self, center):
         coord = self.to_image().get_coord()
         return center.separation(coord.skycoord)
 
-    def get_region_mask_array(self, region):
-        """Return mask of pixels inside region in the the form of boolean array
+    def region_mask(self, regions, inside=False):
+        """Create a mask from a given list of regions
+
+        TODO: implement list of region for each axis
 
         Parameters
         ----------
-        region : `~regions.PixelRegion` or `~regions.SkyRegion` object
-            A region on the sky could be defined in pixel or sky coordinates.
+        regions : list of  `~regions.PixelRegion` or `~regions.SkyRegion` objects
+            A list of regions on the sky (defined in pixel or sky coordinates).
+        inside : bool
+          Output array is set to True inside the input region if inside is set to True and False outside and conversely.
 
         Returns
         -------
-        mask_array : `~numpy.ndarray` of booleans
-            the array of booleans
+        mask_map : `~numpy.ndarray` of boolean type
+            the mask map
+
+        Example
+        -------
+        Example building a mask and storing it in a `~gammapy.maps.WcsNDMap' ::
+            from regions import CircleSkyRegion
+            from astropy.coordinates import SkyCoord, Angle
+            from gammapy.maps import WcsNDMap, WcsGeom
+
+            pos = SkyCoord(0.,0., unit='deg')
+            geom = WcsGeom.create(skydir=pos, npix=100, binsz=0.1)
+
+            pos_region = SkyCoord(3.,3., unit='deg')
+            excluded_region = CircleSkyRegion(pos_region, Angle(0.5, 'deg'))
+
+            # We want to exclude points inside excluded_region
+            mask = geom.region_mask([excluded_region], inside=False)
+
+            # Now we create a map
+            mask_map = WcsNDMap(geom=geom, data=mask)
+            mask_map.plot()
         """
         from regions import PixCoord
 
         if not self.is_regular:
             raise NotImplementedError("Region mask is not working yet with multi-resolution maps")
 
-        # TODO : if Pixel Compound regions are taken into account, rather convert to PixelRegion
-        if isinstance(region, SkyRegion):
-            region = region.to_pixel(self.wcs)
-
         idx = self.get_idx()
         pixcoord = PixCoord(idx[0], idx[1])
-        return region.contains(pixcoord)
+
+        mask = np.zeros(idx[0].shape, dtype=bool)
+
+        for region in regions:
+            # TODO : if Pixel Compound regions are taken into account, rather convert to PixelRegion
+            if isinstance(region, SkyRegion):
+                region = region.to_pixel(self.wcs)
+            mask = np.logical_or(mask, region.contains(pixcoord))
```

Comment:
Isn't this the same, and a little bit less code and faster?
```
mask += region.contains(pixcoord)
```

Reply:
Done